### DISCUSSION TITLE
Confirming 120FPS works

### DISCUSSION THREAD

**Discussion by Cikatriz87** [TheElixZammuto/moonlight-xbox#153] [General]
Confirming 120FPS works. Streaming 1080p120 at 25K bit rate

**Comment by GingerAdonis** [comment]
What about 1440p or 4K on 120Hz? Can anyone confirm that? :)

**Reply by Eamo5** [reply]
4K 120hz confirmed :)

**Reply by GingerAdonis** [reply]
Thank you for your reply. Is that on a Series S or Series X? And would you say the experience is great?

**Reply by Cikatriz87** [reply]
> 4K 120hz confirmed :)

How high was the bit rate you used? I used 40K bit rate with decent results even at 25K for 1080p120. Anything lower I found not enjoyable for 120fps

**Reply by Eamo5** [reply]
Series X with a ethernet cable, though I don't imagine Series S would be much different. I'm maxing out the bitrate at 150Mbps since I don't really see any performance or latency degradation with it maxed out. The weird artifacting that occurs when a portion of the screen has been static is the only real problem I have right now.

**Reply by GingerAdonis** [reply]
And what about surround sound? Does that work properly?

**Reply by Eamo5** [reply]
Not sure, don't have the equipment to validate that just yet

**Reply by plexecutor** [reply]
> And what about surround sound? Does that work properly?

I can confirm surround sound works well. As for 2k120/4k120 I cannot test higher than 1080p120 as my LG B7 only does 60hz at resolution higher than 1080p over HDMI.

**Comment by PythonHugs** [comment]
Just wanted to mention I was using this at 4k 120hz last night and the experience was great

Also was using 7.1 surround at the time

Was on Xbox series x over Ethernet at 150mbps. 

Had a few pauses randomly so probably have my bitrate a bit high... will play with that but moonlight seems good. 

Also will try on Xbox series s later this week but expecting the same.

**Comment by BatJoe** [comment]
1440p 120hz works on my series x in testing.

**Comment by PythonHugs** [comment]
I do get random freezes that last for a second occasionally. When it happens I noticed the logs have a big block of text saying "waiting for idr frame".

I haven't yet figured out how to fix this yet. Not sure if it's a kink somewhere in my setup or an issue with the Xbox moonlight client app. 

Either way 4k 120fps at 150mbps is flawless for me other than this one strange hiccup.

**Comment by BatJoe** [comment]
I am wired locally and at 1440p 120hz 100mbps. I also noticed in some random cases the picture will freeze and go back to normal after 1-2seconds. Not sure why that is. I tried going down to 80mbps and it still happens. So I don't think it is bitrate related.

Maybe this is the same thing @PythonHugs is experiencing.

**Comment by PythonHugs** [comment]
> I am wired locally and at 1440p 120hz 100mbps. I also noticed in some random cases the picture will freeze and go back to normal after 1-2seconds. Not sure why that is. I tried going down to 80mbps and it still happens. So I don't think it is bitrate related.
> 
> 
> 
> Maybe this is the same thing @PythonHugs is experiencing. 

Yes probably. When it happens if you press start+select and then choose show logs I bet you will see the waiting for idr frame stuff. 

It seems the stream gets lost and has to wait a second to pick back up. 

I'm currently messing with my host machine like disabling hardware accelerated gpu scheduling to see if I can find a way to prevent this from happening. The theory being that when gpu goes to 100% utilization the system is pausing the stream to focus on rendering which makes Xbox client have to wait for idr frame. 

If I can get it to stop happening I'll report back. If not, it may be possible that it's some bug in the Xbox client but figuring that out on a c++ project is beyond my skill. 😅

**Reply by BatJoe** [reply]
I should say it doesn't happen on my shield pro. So Im leaning towards app related bug/issue.

**Reply by PythonHugs** [reply]
Ha! Didn't realize you could split off a thread on here. Oops!

Good to know about the shield. If it is an app issue then maybe it can be fixed unless it's a UWP limitation. Hopefully not because that's the only thing that seems to be a downside to using the Xbox client currently. 

At least it seems fairly rare so not the end of the world but if it happens during a critical moment it sure sucks. lol

**Reply by PythonHugs** [reply]
Just wanted to report back. 

I've tried everything and I still get these waiting for idr frame hitches occasionally. Even dropping to 60fps stream didn't stop it so now I'm thinking it's a bug somewhere.

Thankfully it doesn't seem to be too bad but if it happens at the wrong time it can be quite frustrating. In my experience it's the only thing that seems to be holding this back from being basically perfect. 

I did encounter some audio buffer overflows which cause audio dropouts as well but I have a feeling the two issues are actually related.

Hopefully someone smarter than me can figure it all out and push an update. 😅

**Reply by BatJoe** [reply]
I opened a bug issue for it here: https://github.com/TheElixZammuto/moonlight-xbox/issues/261

I have no network issues and this problem only happens on the Moonlight Xbox app. So I suspect something is going on with the app. Changing settings doesn't seem to help. Doesn't happen on the shield pro. There are other weird issues I see with the XBOX app that I don't see either on the Shield Pro. The App will randomly crash sometimes. The image will sometimes have some kind of smudging effect as well. Its too bad cause this could very well be the ultimate Moonlight streaming app as the XBOX is a powerful device.

@andygrundman @TheElixZammuto would of course be the folks to answer to the issues.
